webrtc.html

```diff
@@ -5294,7 +5294,7 @@ <h4>Processing Remote MediaStreamTracks</h4>
           </li>
           <li>
             <p>If <var>track.muted</var> is <code>false</code>,
-            <a>update the muted state</a> of <var>track</var> with the value
+            <a>set the muted state</a> of <var>track</var> to the value
```

Comment:
> The "set muted state" steps are always queued anyhow

@adam-be Here's the one place we don't queue a task (the purpose of this PR), inside the [process the removal of a remote track](http://w3c.github.io/webrtc-pc/#process-remote-track-removal) algorithm which itself is already called from a queued task, to solve https://github.com/w3c/webrtc-pc/issues/1568.

Reply:
If we fire the event immediately, the event will be fired before all effects of the SRD has occurred - and inspecting the PC or related objects like sender/receiver/transceiver/streams/tracks will yield confusing results sensitive to the order of objects processed by this algorithm.

If we schedule a task to fire the event the event will occur after the SRD promise has resolved (which is inconsistent with ontrack for tracks being added) and could theoretically occur at a time when the muted state has already changed again.

We should do what we do for ontrack:
Here: Set the muted state, and if it changed as a result of this, enqueue mutedEvent on to mutedEvents.
Before SRD is resolved, but still synchronous to the rest of the SRD steps: For each mutedEvent in mutedEvents, fire mutedEvent.

It's important that all events fired in relation to an SRD are fired synchronously, but after the state of all objects has been updated, to ensure inspecting the state of the PC is meaningful.